src/main.rs

```diff
@@ -359,8 +362,11 @@ impl Fastmod {
         diffs
     }
 
-    fn print_diff(&mut self, orig: &str, edit: &str) {
+    fn print_diff(&mut self, orig: &str, edit: &str) -> bool {
+        let mut printed_diffs = false;
```

Comment:
nit: it would be nice to avoid mutable state here, since we can. How about `let diffs = self.diffs_to_print(orig, edit);` here, `for diff in diffs {` below, and return `!diffs.is_empty()`?